### PR TITLE
Improve warp drive limit

### LINKED ISSUE #746: Warp disruption distance is too large when moving away celestial bodies
Just like for the planetary rings, ship forward vector should be taken into account to allow warping earlier when leaving a celestial body

### PR DESCRIPTION
**Pull request by BarthPaleologue**
## What does this do?

This PR allows to start warp drive earlier when moving away from a celestial body, and improves the emergency stopping precision by using the altimeter when available.

Bonus: steeper auto lift-off ascent angle to limit edge cases when landed on a slope.

## How to test?

- try to go at full speed toward a planet
- emergency should stop you before colliding
- lift-off from planet
- warp drive should be available sooner

## Interesting findings / surprises / setbacks

Anomalies tend to have non spherical shapes (menger sponge, sierpinski pyramid, mandelbox), for those cases the distance to center mechanism is not the best as coming from the center of a face yeilds a widely different distance to the center than coming from one of the vertices.

It is not obvious how this should be handled, having different distance fields would be clunky, but letting player go at full speed for 30s to be able to warp out of the anomaly is not the correct behavior either.

Maybe having impostors a bit like with the physics engines and doing casts against those shapes could help?

## AI Usage

None

## Related Tickets

Fixes #746 

## What's next?

Find a better solution for non spherical bodies

### CODE REVIEW

File: packages/game/src/ts/frontend/spaceship/components/warpDriveUtils.ts [hunk 1]
```diff
@@ -22,25 +22,37 @@ import { type OrbitalObject } from "@/frontend/universe/architecture/orbitalObje
...
-    const predictedDistanceToObject = Vector3.Distance(
-        predictedPosition,
-        nearestOrbitalObject.getTransform().getAbsolutePosition(),
-    );
-    if (predictedDistanceToObject < emergencyStopDistance) {
+    const predictedTraveledDistance = currentVelocity * nbSecondsPrediction;
+    const predictedDistanceToObject = distanceToObject - predictedTraveledDistance; // worst case scenario
+    const willBeTooClose = predictedDistanceToObject < emergencyMargin;
+    const isGettingCloser = shipForward.dot(directionToObject) > 0;
+    if (willBeTooClose && isGettingCloser) {
```
[the remark this answers is not in the document]
Comment (BarthPaleologue) [reply packages/game/src/ts/frontend/spaceship/components/warpDriveUtils.ts:55]: Not reliable enough, player could still go below surface
